comment/static/js/comment.js

```diff
@@ -351,11 +361,46 @@ document.addEventListener('DOMContentLoaded', () => {
                     fillReaction(parentReactionEle, targetReaction);
                     changeReactionCount(parentReactionEle, data.likes, data.dislikes);
                 }
-            }).catch(error => {
+            }).catch(() => {
                 alert(gettext("Reaction couldn't be processed!, please try again"));
             });
         };
 
+    let toggleFollow = (followButton, form) => {
+        let formDataQuery = null;
+        if (form) {
+            let formData = serializeObject(form);
+            formDataQuery = convertFormDataToURLQuery(formData);
+        }
+        let url = followButton.getAttribute('data-url');
+        fetch(url,  {
+            method: 'POST',
+            headers: headers,
+            body: formDataQuery,
+        }).then(response => {
+            return response.json();
+        }).then(data => {
+            if (data.email_required) {
+                followModal.querySelector('form').setAttribute('data-target-btn-id', followButton.getAttribute('id'));
+                showModal(followModal);
+            }
+            else if (data.invalid_email) {
+                form.querySelector('.error').innerHTML = data.invalid_email;
+            }
+            else if (data.following) {
+                followButton.querySelector('.comment-follow-icon').classList.add('user-has-followed');
+                followButton.querySelector('span').setAttribute('title', 'Unfollow this thread');
+                hideModal(followModal);
+            } else {
+                followButton.querySelector('.comment-follow-icon').classList.remove('user-has-followed');
+                followButton.querySelector('span').setAttribute('title', 'Follow this thread');
```

Comment:
In addition to this change of icons, I think we can also, use the function `createInfoElement` to display a message describing to the user whether they have subscribed/unsubscribed successfully. At the moment, after clicking on the follow icon, i missed the slight change in its color when this request completed.